freva-rest/src/databrowser_api/core.py

```diff
@@ -1127,356 +1122,175 @@ def escape_special_chars(value: str) -> str:
 
         Parameters
-        ~~~~~~~~~~
-        path : os.PathLike
-            The path to either a file or a directory containing user data.
+        ----------
+        user_metadata : List[Dict[str, Any]]
+            A list of user metadata to be processed.
 
         Returns
-        ~~~~~~~
+        -------
         None
         """
-        paths_to_process = []
-        if Path(path).is_file() and any(
-            Path(path).suffix == suffix for suffix in self.suffixes
-        ):
-            paths_to_process.append(path)
-        elif Path(path).is_dir():
-            async for file_path in self._gather_files_from_dir(Path(path)):
-                paths_to_process.append(file_path)
-                if len(paths_to_process) >= self.batch_size:
-                    future = self.executor.submit(
-                        run_async_in_thread,
-                        self._process_paths_in_executor,
-                        paths_to_process,
-                        path,
-                    )
-                    self.submitted_tasks_in_excutor.append(future)
-                    paths_to_process = []
+        user_metadata_batch_to_process = []
+        for metadata in user_metadata:
+            metadata.update(self.fwrites)
+            user_metadata_batch_to_process.append(metadata)
+            if len(user_metadata_batch_to_process) >= self.batch_size:
+                self.executor.submit(
+                    run_async_in_thread,
+                    self._process_user_metadata_in_executor,
+                    user_metadata_batch_to_process,
+                )
+                user_metadata_batch_to_process = []
 
-        if paths_to_process:
-            future = self.executor.submit(
-                run_async_in_thread, self._process_paths_in_executor,
-                paths_to_process, path,
+        if user_metadata_batch_to_process:
+            self.executor.submit(
+                run_async_in_thread,
+                self._process_user_metadata_in_executor,
+                user_metadata_batch_to_process,
             )
-            self.submitted_tasks_in_excutor.append(future)
-        self.get_executor_info()
-
-    async def _gather_files_from_dir(self, path: Path) -> AsyncIterator[Path]:
-        """
-        Gather files from a directory based on specific patterns and suffixes.
-
-        Parameters
-        ~~~~~~~~~~
-        path : Path
-            The directory path to scan for files.
-
-        Yields
-        ~~~~~~
-        Path
-            Files that match the given suffix patterns.
-        """
-        for item in Path(path).rglob("*.*"):
-            if item.is_file() and any(
-                item.suffix == suffix for suffix in self.suffixes
-            ):
-                yield item
 
-    async def _process_paths_in_executor(
-            self, paths: List[Path], path_to_mongo: Path) -> None:
+    async def _process_user_metadata_in_executor(self,
```

Comment:
Do we really need that function? I understand that it was there when you actually opened files to get the metadata. This could have been a potentially long process. But since we have all metadata aboard I think we don't need to run this in a different thread at all. Adding the data to solr and mongo should be fast.